### ISSUE TITLE
No ICD Detected when primus installed

### ISSUE THREAD

**Issue by lschekotihin** [lutris/lutris#4292]
### Bug description

I am trying to run a game on my laptop with optimus.
Primary Nvidia, secondary AMD. 
As a runner i have intended to use Primus with non free driver on Debian. Demos and some standalone application works fine.
When i have opened game options i cant find Nvidia in ICDs dropdown. 
ICD file was in place so i have looked at lutris code : [elif "nvidia" in loader:](https://github.com/lutris/lutris/blob/9f7f90a6017c3abe2584bfb7f920e8d9cf92863b/lutris/sysoptions.py#L97)

In me case the naming of files have changed due to primus installed:

```
drwxr-xr-x 2 root root 4,0K мая 17 12:59 .
drwxr-xr-x 5 root root 4,0K мая 12 21:47 ..
-rw-r--r-- 1 root root  161 апр 10 10:18 intel_icd.i686.json
-rw-r--r-- 1 root root  163 апр 10 10:18 intel_icd.x86_64.json
-rw-r--r-- 1 root root  159 апр 10 10:18 lvp_icd.i686.json
-rw-r--r-- 1 root root  161 апр 10 10:18 lvp_icd.x86_64.json
-rw-r--r-- 1 root root  140 янв  6 15:10 nvidia_icd.disabled_by_nv_vulkan_wrapper
-rw-r--r-- 1 root root  146 апр  4 00:30 nv_vulkan_wrapper.json
-rw-r--r-- 1 root root  162 апр 10 10:18 radeon_icd.i686.json
-rw-r--r-- 1 root root  164 апр 10 10:18 radeon_icd.x86_64.json
```

So Nvidia driver was not found literally.

### How to Reproduce

1. Install latesr primus on Nvidia laptop
2. Open game options and look for found ICDs

### Expected behavior

Driver names imposed by primus can be found.

### Log output

```shell
---
```

### System Information

```shell
Lutris 0.5.10.1
Primus version "0~20150328-14"
primus_vk "1.6.1-2"

[System]
OS:              Debian GNU/Linux testing bookworm
Arch:            x86_64
Kernel:          5.17.0-1-amd64
Desktop:         XFCE
Display Server:  x11

[CPU]
Vendor:          AuthenticAMD
Model:           AMD Ryzen 7 5800HS with Radeon Graphics
Physical cores:  8
Logical cores:   16

[Memory]
RAM:             15.1 GB
Swap:            1.0 GB

[Graphics]
Vendor:          AMD
OpenGL Renderer: AMD RENOIR (DRM 3.44.0, 5.17.0-1-amd64, LLVM 13.0.1)
OpenGL Version:  4.6 (Compatibility Profile) Mesa 21.3.8
OpenGL Core:     4.6 (Core Profile) Mesa 21.3.8
OpenGL ES:       OpenGL ES 3.2 Mesa 21.3.8
Vulkan:          Supported
```

### Checklist:

- [X] I'm not asking for support with a game or the wine runner.
- [X] I have followed the above mentioned guides and have all the graphics and wine dependencies installed.
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by jmbreuer** [comment]
I'm experiencing what appears to be at the core the same problem - and this issue report came up at the top of my searches.

What started out as a bit of comment bit by bit turned into a full-blown lab log - I'll just leave this here in the hopes that it may help others, and eventually get the road to success a bit less bumpy.

I'd normally use `pvkrun whatever` or `primusrun whatever` to use my discrete GPU via Primus, depending on whether it's a Vulkan or OpenGL based 3D Application.

I'm running lutris 0.5.11 on gentoo Linux x86_64; on a laptop with discrete nVidia GeForce GTX 950M and integrated Intel HD Graphics 530 (SKL GT2) graphics. Display is connected to the Intel graphics only, i.e. the nVidia GPU provides rendering offload only.

On this system, Vulkan needs to be able to load *both* ICDs - the Intel one for the 'Display GPU', and the nVidia one for the 'Render GPU'. `pvkrun` mostly takes care of enabling/disabling the nVidia GPU using `bbswitch`, the nVidia ICD will report itself as available or not accordingly. 

Things like `pvkrun vkcube` or `primus stellarium` work as expected on my system, as well as e.g. running `vkcube` by itself (when it will use the integrated Intel graphics, or apparently the nVidia GPU as well while it's bbswitch'd on).

Is there a guide how to set up lutris on such a system? I couldn't find any, and comprehensive docs on how all the different parts interact seems scarce even for getting discrete rendering working with native applications, let alone including the whole wine/lutris/etc stack. Following up on the issues I was having, I wound up here as one of the first places.


To keep the stack more simple, quite at first I noticed `gamemoderun` being used, so I disabled Feral GameMode for now - another component that injects code/environment. Something to try out / turn on again when the rest of this Works(TM).

I started on this journey to play a game I recently got for free on the Epic store. In light of this journey's bumpiness, I'd rather try things out with something small and simple first - any suggestions for a suitable game that will allow me to test the 3D setup and is a straightforward & not huge install?


Back to said journey:

In the advanced options, I see a drop-down which should allow me to choose an ICD - however, this seems to use pre-configured values, not all ICDs available on my system. "Out of the box" Lutris will try to use the nVidia ICD on my system, while the Intel one is suppressed with the `VK_ICD_FILENAMES` environment variable that Lutris sets up:

```
PrimusVK: Searching for display GPU:
PrimusVK: 0x561b041ffb80: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x561b041ffb80.
PrimusVK: 0x561b044defd0.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
lutris-wrapper: Epic Games Store
Started initial process 74079 from pvkrun /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/bin/wine /home/jmbreuer/LOCAL/Lutris/epic-games-store/drive_c/Program Files (x86)/Epic Games/Launcher/Portal/Binaries/Win32/EpicGamesLauncher.exe -opengl -SkipBuildPatchPrereq
Start monitoring process.
esync: up and running.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7f729140: 4318;5018
PrimusVK: Searching for render GPU:
PrimusVK: 0x7f729140.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: No device for the display GPU found. Are the intel-mesa drivers installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json
```

This led me to try to specify a sensible value for `VK_ICD_FILENAMES` manually using the Environment Variable thing on the Wine runner's System Settings tab in Lutris.

At first, I tried with `/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json`, but I still couldn't successfully run anything really wanting to use 3D/discrete graphics, starting with the Epic Games Store. It launches it's "Updating..." splash screen, and once that's done it quits with an error or an "Unsupported Graphics Gard" dialog from wine (showing with the wine glass app icon):

```
lutris-wrapper: Epic Games Store
Started initial process 76711 from pvkrun /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/bin/wine /home/jmbreuer/LOCAL/Lutris/epic-games-store/drive_c/Program Files (x86)/Epic Games/Launcher/Portal/Binaries/Win32/EpicGamesLauncher.exe -opengl -SkipBuildPatchPrereq
Start monitoring process.
esync: up and running.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7eaa09d0: 4318;5018
PrimusVK: 0x7eaa8c50: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7eaa09d0.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: Surpressing GetRandROutputDisplay
PrimusVK: Surpressing GetRandROutputDisplay
LogInit: Display: Command Line: "C:\Program Files (x86)\Epic Games\Launcher\Portal\Binaries\Win32\EpicGamesLauncher.exe" -opengl -SkipBuildPatchPrereq -SaveToUserDir -Messaging
LogStreaming: Warning: Failed to read file '../../../Engine/Content/Slate/Old/DashedBorder.png' error.
LogStreaming: Warning: Failed to read file '../../../Engine/Content/Slate/Icons/cursor_grab.png' error.
[... tons more of these redacted for readability...]
LogStreaming: Warning: Failed to read file '../../Content/New UI/Social/TextBox_ReadOnly.png' error.
LogStreaming: Warning: Failed to read file 'C:/ProgramData/Epic/EpicGamesLauncher/Data/Launcher.manifest' error.
LogWindows: Warning: Register touch input failed!
Slate User Destroyed.  User Index 0, Is Virtual User: 0
Initial process has exited (return code: 0)
PrimusVK: Searching for display GPU:
PrimusVK: 0xf40e2030: 4318;5018
PrimusVK: Searching for render GPU:
PrimusVK: 0xf40e2030.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: No device for the display GPU found. Are the intel-mesa drivers installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0xf42429b0: 4318;5018
PrimusVK: Searching for render GPU:
PrimusVK: 0xf42429b0.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: No device for the display GPU found. Are the intel-mesa drivers installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
XIO:  fatal IO error 4 (Interrupted system call) on X server ":8"
      after 91 requests (91 known processed) with 0 events remaining.
Monitored process exited.
Exit with return code 0
2022-10-21 13:32:12,735: Game still running (state: running)
2022-10-21 13:32:12,736: Stopping Epic Games Store (wine)
```

"Something" gets broken passing a usable graphics context originating from `pvkrun` into applications launched from Lutris.

I've played around starting Lutris itself with `pvkrun` (and then setting the options to not use pvkrun), but basically, it leads to the same results - i.e. lutris seems to be doing a good job of isolating the programs it launches from its own environment ;-)

The log above is with 'Enable NVIDIA Prime Offload' and 'Use discrete graphics' both enabled. What do these options do exactly?

With only Prime Offload enabled, I get identical console output, but the splash window flickers badly to black and doesn't appear to update properly.

With only discrete graphics enabled, or both options turned off, the end of the log looks slightly different:
```
lutris-wrapper: Epic Games Store
Started initial process 77338 from pvkrun /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/bin/wine /home/jmbreuer/LOCAL/Lutris/epic-games-store/drive_c/Program Files (x86)/Epic Games/Launcher/Portal/Binaries/Win32/EpicGamesLauncher.exe -opengl -SkipBuildPatchPrereq
Start monitoring process.
esync: up and running.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7e7beef0: 4318;5018
PrimusVK: 0x7e7bee20: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7e7beef0.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: Surpressing GetRandROutputDisplay
PrimusVK: Surpressing GetRandROutputDisplay
LogInit: Display: Command Line: "C:\Program Files (x86)\Epic Games\Launcher\Portal\Binaries\Win32\EpicGamesLauncher.exe" -opengl -SkipBuildPatchPrereq -SaveToUserDir -Messaging
LogStreaming: Warning: Failed to read file '../../../Engine/Content/Slate/Old/DashedBorder.png' error.
LogStreaming: Warning: Failed to read file '../../../Engine/Content/Slate/Icons/cursor_grab.png' error.
LogStreaming: Warning: Failed to read file '../../../Engine/Content/Slate/Icons/cursor_cardinal_cross.png' error.
[...]
LogStreaming: Warning: Failed to read file '../../Content/New UI/Social/TextBox_Hovered.png' error.
LogStreaming: Warning: Failed to read file '../../Content/New UI/Social/TextBox_ReadOnly.png' error.
LogStreaming: Warning: Failed to read file 'C:/ProgramData/Epic/EpicGamesLauncher/Data/Launcher.manifest' error.
LogWindows: Warning: Register touch input failed!
X Error of failed request:  GLXBadDrawable
  Major opcode of failed request:  151 (GLX)
  Minor opcode of failed request:  11 (X_GLXSwapBuffers)
  Serial number of failed request:  33
  Current serial number in output stream:  34
```
... and the Epics Game Store hangs there with an invisible window. I can kill it from the task bar.

With `pvkrun` turned off in Lutris' wine options *and* explicitly setting the Intel ICD (and getting my `VK_ICD_FILENAMES` override out of the way), I could finally launch the EGS and see its interface at all. But obviously it will be using the integrated graphics unsuitable for any (3D) game.

I tried installing and running Tacoma at this point. Installing it works fine, but it's not exactly playable on my Intel graphics.

Around this point I found that clicking on the 'play' icon next to the EGS in the left-hand list does not necessarily seem to do exactly the same thing as double-clicking on the EGS banner within the Wine runner section - I'd previously always done the above, assuming that games I'd install within EGS would show up with separate launchers in its section - ah, apparently that's issue #4472. I also noticed that some of the Optimus/Prime/discrete graphics options not only exist in the context of the Wine runner, but also in the Global Options in the hamburger menu. At that point I set those to the same so-far best-working values, sans the `VK_ICD_FILENAMES` override.

Rather by accident at that point I launched the EGS from the Wine runner with pvkrun/Primus/discrete/my ICD env override enabled, and I got different output:

```
lutris-wrapper: Epic Games Store
Started initial process 86166 from pvkrun /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/bin/wine /home/jmbreuer/LOCAL/Lutris/epic-games-store/drive_c/Program Files (x86)/Epic Games/Launcher/Portal/Binaries/Win32/EpicGamesLauncher.exe -opengl -SkipBuildPatchPrereq
Start monitoring process.
esync: up and running.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7fbf7de0: 4318;5018
PrimusVK: 0x7fc00290: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7fbf7de0.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: Surpressing GetRandROutputDisplay
PrimusVK: Surpressing GetRandROutputDisplay
Initial process has exited (return code: 0)
Can't open bumblebee display.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7ec09900: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7ec09900.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0x7edb6890: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7edb6890.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0x7ebfc360: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7ebfc360.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
wine: Unhandled exception 0x20474343 in thread 1dc at address 000000007B0124DE (thread 01dc), starting debugger...
Can't open bumblebee display.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7e868ed0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7e868ed0.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
Can't open bumblebee display.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7ea36c40: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7ea36c40.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0x7eb7ca80: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7eb7ca80.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0x7e9434f0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7e9434f0.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
wine: Unhandled exception 0x20474343 in thread 254 at address 000000007B0124DE (thread 0254), starting debugger...
Can't open bumblebee display.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7f4c6600: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7f4c6600.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0x7f6728e0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7f6728e0.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
PrimusVK: Searching for display GPU:
PrimusVK: 0x7f4c10e0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7f4c10e0.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
wine: Unhandled exception 0x20474343 in thread 340 at address 000000007B0124DE (thread 0340), starting debugger...
Can't open bumblebee display.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7f1effb0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7f1effb0.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nv_vulkan_wrapper.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json

(wine:86410): GStreamer-WARNING **: 15:53:41.094: Failed to load plugin '/home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/lib64/gstreamer-1.0/libgstspeex.so': libspeex.so.1: cannot open shared object file: No such file or directory

(wine:86410): GStreamer-WARNING **: 15:53:41.094: Failed to load plugin '/home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/lib64/gstreamer-1.0/libgstflac.so': libFLAC.so.8: cannot open shared object file: No such file or directory

(wine:86410): GStreamer-WARNING **: 15:53:41.096: Failed to load plugin '/home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/lib/gstreamer-1.0/libgstspeex.so': /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/lib/gstreamer-1.0/libgstspeex.so: wrong ELF class: ELFCLASS32

(wine:86410): GStreamer-WARNING **: 15:53:41.096: Failed to load plugin '/home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/lib/gstreamer-1.0/libgstflac.so': /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/lib/gstreamer-1.0/libgstflac.so: wrong ELF class: ELFCLASS32
```

So at that point, it "saw" both my display (Intel) and render (nVidia) GPUs, and this time, it couldn't load the ICD for the nVidia one. Hmm, try the proper nVidia ICD instead of the primus_vk wrapper/shim? => `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json`

Lo and behold, EGS would launch with its UI rendered by the nVidia GPU (I use `intel_gpu_top` and `nvidia-smi` beyond appearance and general sluggishness of my desktop to check).

Thus encouraged, I tried Tacoma, and it launches in a playable fashion utilizing the nVidia GPU.

Besides both hunch and analysis pointing to how relevant settings might interact between their Global options and Wine runner System settings instances, unfortunately I don't fully understand what made it work in the end. But I can check/try things in my now-running setup, if that helps.

Now I only wish I could easily switch one same runner between different configurations - to download/install EGS games with the nVidia space heater (and fan accelerator) disabled, and then play those with the nVidia GPU enabled. I guess that's might actually be supposed to work in another way, and it might be issue #4472 preventing this for me - I don't get launchers for individual EGS games, just the EGS itself as a launcher. Hm, shouldn't this already be fixed in 0.5.11?  - I'll continue this in its own appropriate issue.

**Comment by danieljohnson2** [comment]
TLDR, man.

But we've done some improvements in the handling of hybrid graphics since 0.5.11; could you try the [just-released beta of 0.5.12](https://github.com/lutris/lutris/releases/tag/v0.5.12-beta1), and see if that helps?

**Comment by jmbreuer** [comment]
Before reading your reply, I'd already switched to a lutris install tracking the git master. Hope this is OK to analyze further, too. It reports as 0.5.12, and is currently based on commit 2bd7d87ee6652ea4c64575cbb33342abe8b6386f

As it is, it doesn't behave immediately differently from my experiences with 0.5.11.

Just configuring for `pvkrun` causes discrete graphics to be "lost" again early:

```
Started initial process 134651 from pvkrun /home/jmbreuer/.local/share/lutris/runners/wine/lutris-7.2-2-x86_64/bin/wine /home/jmbreuer/LOCAL/Lutris/epic-games-store/drive_c/Program Files (x86)/Epic Games/Launcher/Portal/Binaries/Win32/EpicGamesLauncher.exe -opengl -SkipBuildPatchPrereq -com.epicgames.launcher://apps/Flagfin?action=launch
Start monitoring process.
esync: up and running.
PrimusVK: Searching for display GPU:
PrimusVK: 0x7ef604a0: 4318;5018
PrimusVK: 0x7ef5c4f0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7ef604a0.
PrimusVK: Got discrete gpu!
PrimusVK: Device: NVIDIA GeForce GTX 950M
PrimusVK:   Type: 2
PrimusVK: Surpressing GetRandROutputDisplay
PrimusVK: Surpressing GetRandROutputDisplay
Initial process has exited (return code: 0)
PrimusVK: Searching for display GPU:
PrimusVK: 0x7ddba900: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7ddba900.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
[ This log section repeats numerous times: ]
[ === vvvv  === ]
PrimusVK: Searching for display GPU:
PrimusVK: 0x7df095e0: 32902;6427
PrimusVK: Got integrated gpu!
PrimusVK: Device: Intel(R) HD Graphics 530 (SKL GT2)
PrimusVK:   Type: 1
PrimusVK: Searching for render GPU:
PrimusVK: 0x7df095e0.
PrimusVK: No device for the rendering GPU found. Is the correct driver installed?
PrimusVK: VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json:/usr/share/vulkan/icd.d/intel_icd.x86_64.json:/usr/share/vulkan/icd.d/intel_icd.i386.json
[ === ^^^^ === ]
```

Remembering what might have been different when it eventually worked (better) for me on Friday, I have stumbled across a workaround. On Friday, I'd probably left `nvidia-smi` running at some point, which means the discrete GPU remains bbswitch'd on, and its kernel module loaded.

I did just this today - ran `pvkrun vkcube`, started `nvidia-smi -l 2`, quit `vkcube`.

When I _then_ run Tacoma through lutris, it picks up and uses the discrete graphics.

Which indicates to me: When launching Tacoma through lutris, configured with `pvkrun`, some processes fork off "to the side", and the process started within `pvkrun` terminates fairly early (at the point where the console says `Initial process has exited (return code: 0)`). Which, at this point, causes `pvkrun` to turn off the discrete graphics again. Thus the issue is probably timing-sensitive - if whatever got launched "to the side" already initialized its own Vulkan library, it would not be possible for `pvkrun` to release/bbswitch-off the discrete GPU.
In my example above, the next Vulkan init only happens shortly afterwards - the next `PrimusVK` stanza, which "correctly" reports the discrete GPU unavailable. 

I'll play around with this and lutris' settings some more and report on what more I figure out.

**Comment by danieljohnson2** [comment]
OK,  OK, I'll ready just a little of that massive thing.

Looks like you suspect a particular check of depriving you of NVIDIA support because of odd file naming. I can alter that check; please try [this branch](https://github.com/lutris/lutris/tree/dj/nv_for_nvidia). You need not install it; just download it and run the "bin/lutris" script in it.

If it fixes your problem, I will merge this for 0.5.12.

**Comment by danieljohnson2** [comment]
A heart emoji, but no response?

This branch is rather out of date now. Maybe I should just delete it.

**Comment by lschekotihin** [comment]
> 

My problem is currently resolved as i have updated my distro and last optimus is just wors, and no more need gor ICDs, run arguments and ETC
